akka-docs/src/main/paradox/cluster-client.md

```diff
@@ -227,3 +232,62 @@ This can be useful when initial contacts are provided from some kind of service
 are entirely dynamic and the entire cluster might shut down or crash, be restarted on new addresses. Since the
 client will be stopped in that case a monitoring actor can watch it and upon `Terminate` a new set of initial
 contacts can be fetched and a new cluster client started.
+
+## Migration to Akka gRPC
+
+Cluster Client is deprecated and it is not advised to build new applications with Cluster Client.
+We recommend using [Akka gRPC](https://doc.akka.io/docs/akka-grpc/current/index.html) over using Cluster Client.
+
+Cluster Client is convenient to use if you have Akka actors on both client and server side, but
+a more decoupled, and therefore better, solution would be to use Akka gRPC and define
+application specific protocol buffer messages and gRPC service calls. The benefits would be:
+
+* Easier to update clients and servers independent of each other.
+* More well defined protocol between client and server.
+* More secure solution by using TLS for gRPC (HTTP/2) instead of exposing Akka Remoting outside the Akka Cluster.
```

Comment:
I'd bump this to the top bullet

Reply:
I was also thinking that service discovery could be a pro

Reply:
Definitely a pro.